frappe/public/js/frappe/web_form/web_form_class.js

```diff
@@ -0,0 +1,124 @@
+frappe.provide("frappe.ui");
+
+window.web_form = null;
+
+
+frappe.ui.WebForm = class WebForm extends frappe.ui.FieldGroup {
+	constructor(opts) {
+		super();
+		Object.assign(this, opts);
+		window.web_form = this;
+	}
+
+	make(opts) {
+		super.make();
+		this.set_field_values();
+		if (this.allow_delete) this.setup_delete_button();
+		this.setup_primary_action();
+		$('.link-btn').remove()
+	}
+
+	set_field_values() {
+		if (this.doc_name) this.set_values(this.doc);
+		else return;
+	}
+
+	setup_primary_action() {
+		const primary_action_button = document.createElement("button");
+		primary_action_button.classList.add(
+			"btn",
+			"btn-primary",
+			"primary-action",
+			"btn-form-submit",
+			"btn-sm",
+			"ml-2"
+		);
+		primary_action_button.innerText = web_form.button_label || "Save";
+		primary_action_button.onclick = () => this.save();
+		document
+			.querySelector(".web-form-actions")
+			.appendChild(primary_action_button);
+	}
+
+	setup_delete_button() {
+		const delete_button = document.createElement("button");
```

Comment:
Why avoid jQuery? It's not going anywhere anytime soon..

Reply:
No particular reason, had slightly better experience with native API and it's mostly inter-operable.